src/Darryldecode/Cart/Cart.php

```diff
@@ -641,11 +640,15 @@ protected function validate($item)
      */
     protected function addRow($id, $item)
     {
+        $this->events->fire($this->getInstanceName().'.adding', array($item, $this));
+
         $cart = $this->getContent();
 
         $cart->put($id, new ItemCollection($item));
 
         $this->save($cart);
+
+        $this->events->fire($this->getInstanceName().'.added', array($item, $this));
     }
 
     /**
```

Comment:
Hey, Could you tell me when and how to contribute to any project?

There were no problems reported, But I had the issue that as when a condition of an item was changed no event get triggered. In place of the triggering the update when adding an item, it seems better to do this inside the update function itself.